src/index.mjs

```diff
@@ -63,40 +43,89 @@ function isIgnored(node, filter) {
  * DOM element or static array of URLs.
  * @param {Object} options - Configuration options for quicklink
- * @param {Array} options.urls - Array of URLs to prefetch (override)
- * @param {Object} options.el - DOM element to prefetch in-viewport links of
- * @param {Boolean} options.priority - Attempt higher priority fetch (low or high)
- * @param {Array} options.origins - Allowed origins to prefetch (empty allows all)
- * @param {Array|RegExp|Function} options.ignores - Custom filter(s) that run after origin checks
- * @param {Number} options.timeout - Timeout after which prefetching will occur
- * @param {Function} options.timeoutFn - Custom timeout function
+ * @param {Object} [options.el] - DOM element to prefetch in-viewport links of
+ * @param {Boolean} [options.priority] - Attempt higher priority fetch (low or high)
+ * @param {Array} [options.origins] - Allowed origins to prefetch (empty allows all)
+ * @param {Array|RegExp|Function} [options.ignores] - Custom filter(s) that run after origin checks
+ * @param {Number} [options.timeout] - Timeout after which prefetching will occur
+ * @param {Number} [options.throttle] - The concurrency limit for prefetching
+ * @param {Number} [options.limit] - The total number of prefetches to allow
+ * @param {Function} [options.timeoutFn] - Custom timeout function
  */
-export default function (options) {
+export function listen(options) {
   if (!options) options = {};
+  if (!window.IntersectionObserver) return;
+
+  const [toAdd, isDone] = throttle(options.throttle || 1/0);
+  const limit = options.limit || 1/0;
 
-  observer && (observer.priority = options.priority || false);
+  // TODO: I think this isn't needed?
+  const isPriority = !!options.priority;
 
   const allowed = options.origins || [location.hostname];
   const ignores = options.ignores || [];
 
-  const timeout = options.timeout || 2e3;
   const timeoutFn = options.timeoutFn || requestIdleCallback;
 
-  timeoutFn(() => {
-    // If URLs are given, prefetch them.
-    if (options.urls) {
-      options.urls.forEach(prefetcher);
-    } else if (observer) {
-      // If not, find all links and use IntersectionObserver.
-      Array.from((options.el || document).querySelectorAll('a'), link => {
-        observer.observe(link);
-        // If the anchor matches a permitted origin
-        // ~> A `[]` or `true` means everything is allowed
-        if (!allowed.length || allowed.includes(link.hostname)) {
-          // If there are any filters, the link must not match any of them
-          isIgnored(link, ignores) || toPrefetch.add(link.href);
+  const observer = new IntersectionObserver(entries => {
+    entries.forEach(entry => {
+      if (entry.isIntersecting) {
+        observer.unobserve(entry = entry.target);
+        // Do not prefetch if will match/exceed limit
+        if (toPrefetch.size < limit) {
+          toAdd(() => {
+            // TODO: Don't need isPriority?
+            prefetch(entry.href, isPriority).then(isDone);
```

Comment:
Would be good to add a `catch` handler here now; exposed as `options.onError` perhaps? https://github.com/GoogleChromeLabs/quicklink/pull/70#issuecomment-482699527

This would still allow/require user to do custom error handling when calling `quicklink.prefetch` directly.

Reply:
> Would be good to add a catch handler here now; exposed as options.onError perhaps?

SGTM and thank you for circling back on #70. Even though we would be deferring to users for custom error handling in `.prefetch()`, adding a `catch` -> onError is a good call here.